tests/lib/helper.ts

```diff
@@ -1,15 +1,29 @@
 import { expect,Page } from '@playwright/test';
 
 export const waitForCustomEvent = async(page:Page, customEventName:string) => {
-    return page.evaluate(eventName => new Promise(callback => window.addEventListener(eventName, callback,{ once: true })), customEventName)
+  return page.evaluate(eventName => new Promise(callback => window.addEventListener(eventName, callback,{ once: true })), customEventName)
 }
 
 export const waitForCustomEventTimeout = async (page:Page,customEventName:string) => {
-    const value = await Promise.race([
-        page.evaluate(eventName => new Promise(callback => window.addEventListener(eventName, callback, { once: true })), customEventName),
-        new Promise((resolve) => {setTimeout(resolve, 500, "timeout")}),
-    ])
+  const value = await Promise.race([
+    page.evaluate(eventName => new Promise(callback => window.addEventListener(eventName, callback, { once: true })), customEventName),
+    new Promise((resolve) => {setTimeout(resolve, 500, "timeout")}),
+  ])
 
-    expect(value).toBe("timeout")
+  expect(value).toBe("timeout")
 }
 
+// This function adds an event listener for eventName and puts the value of paramName on the window
+// Retrieve the value of paramName by calling await page.evaluate('window.<paramName>')
+export const waitForCustomEventWithParam = (page:Page, eventName:string, paramName:string) => {
```

Comment:
Are there any concerns with placing the value on the window causing issues between tests? is there any clean up we could do of the value afterwards so another test doesnt pick it up?

Reply:
I think if the tests are broken down into one event of each type per test, we shouldn't have any problems (every `page.goto()` should create a new page environment), though I should test this function more. I imagine this would cause a problem if one test dispatches for example, two "input" events with the same value.

If that's causing a problem, maybe we can also create a helper function `getCustomEventValue` to retrieve the event value (`page.evaluate('window.eventName')`) and also clear it from the window?